Bearded.Utilities.Tests/Generators/DirectionGenerators.cs

```diff
@@ -0,0 +1,14 @@
+using Bearded.Utilities.Geometry;
+using FsCheck;
+
+namespace Bearded.Utilities.Tests.Generators
+{
+    static class DirectionGenerators
+    {
+        public static class All
+        {
+            public static Arbitrary<Direction2> Directions() =>
+                Arb.Generate<uint>().Select(i => Direction2.FromDegrees(360f / (1L << 32))).ToArbitrary();
```

Comment:
This makes me realise that it may be impossible to easily construct many possible Direction2 values since floats have less significant digits than the underlying integer. Curious.

Reply:
Yes, I am not sure if it makes sense to expose the underlying data model, but you can see here that that leads to us converting back and forth. There doesn't seem to be an easy way to easily generate floats in a certain range without just filtering, which may throw away edge cases.

Reply:
edge cases = test cases